src/main/java/mate/academy/dao/impl/MovieDaoImpl.java

```diff
@@ -47,12 +47,60 @@
     @Override
     public List<Movie> getAll() {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            CriteriaQuery<Movie> criteriaQuery = session.getCriteriaBuilder()
-                    .createQuery(Movie.class);
-            criteriaQuery.from(Movie.class);
-            return session.createQuery(criteriaQuery).getResultList();
+            return session.createQuery("from Movie").list();
         } catch (Exception e) {
-            throw new DataProcessingException("Can't get all movies", e);
+            throw new DataProcessingException("Can't get Movies", e);
         }
     }
+
+    @Override
+    public boolean update(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.update(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("updating " + movie + " failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+
+        return get(movie.getId()).get().equals(movie);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.delete(get(id)
+                    .orElseThrow(() -> new RuntimeException("no object with id: " + id)));
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("removing movie with id: "
+                    + id + " from database failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return get(id).isEmpty();
```

Comment:
The `delete` method checks if the object is deleted by verifying if the object is absent in the database. This is a valid approach, but ensure that the `get` method correctly handles the absence of an object to avoid unnecessary exceptions.